### ISSUE TITLE
provide gradient function for SoftMax and Sigmoid in System.Numerics.Tensors.TensorPrimitives

### ISSUE THREAD

**Issue by BarionLP** [dotnet/runtime#109328]
It would be really nice if System.Numerics.Tensors.TensorPrimitives had partial derivative functions (gradient) for SoftMax and Sigmoid as they are commonly needed in machine learing.

**Comment by lilinus** [comment]
What do you mean with derivative for SoftMax, since it is a vector-valued multivariate function? The Jacobian matrix?

**Comment by BarionLP** [comment]
> What do you mean with derivative for SoftMax, since it is a vector-valued multivariate function? The Jacobian matrix?

the pointwise derivative with respect to that element (I think that what's it called) used in backpropagation.
$$∂softmax(z)_i \over ∂z_j$$

**Comment by BarionLP** [comment]
thats what I'm using in my little experiment.
https://gist.github.com/BarionLP/03f7dcd61d5d604a5ef36e303f990120
(I'm not an expert)

**Comment by jeffhandley** [comment]
Marking as [help wanted](https://github.com/dotnet/runtime/labels/help%20wanted) to invite a concrete API proposal, ideally with a link to a prototype implementation. The [api-proposal skill](https://github.com/dotnet/runtime/tree/main/.github/skills/api-proposal) might be handy for someone to use for this.
